tests/product/test_server_upgrade.py

```diff
@@ -107,3 +107,28 @@ def test_rolling_upgrade(self):
     def copy_upgrade_rpm_to_cluster(self):
         rpm_name = self.dummy_installer.copy_presto_rpm_to_master()
         return os.path.join(self.cluster.mount_dir, rpm_name)
+
+    def test_upgrade_fails_given_directory(self):
+        dir_on_cluster = '/opt/prestoadmin'
+        self.assertRaisesRegexp(
+            OSError,
+            'RPM file not found at %s. Make sure to specify the filename'
+            'and not the directory name' % dir_on_cluster,
+            self.run_prestoadmin,
+            'server upgrade ' + dir_on_cluster
+        )
+
+    def test_upgrade_works_with_symlink(self):
+        self.start_and_assert_started()
+
+        self.run_prestoadmin('configuration deploy')
+        for container in self.cluster.all_hosts():
+            self.real_installer.assert_installed(self, container)
+            self.assert_has_default_config(container)
```

Comment:
Shouldn't we assert some different configuration?
Even if upgrade overwrites configuration, it would overwrite it to default, so this test does not check if we're preserving configuration on upgrade.